apps/gallery/js/ImageEditor.js

```diff
@@ -124,6 +124,11 @@ function editPhoto(n) {
     $('edit-tool-apply-button').disabled = true;
 
     window.addEventListener('resize', resizeHandler);
+    // If the locale changes while user is using exposure edit option,
+    // then we need to re-position slider thumb for respective locale.
+    navigator.mozL10n.ready(function() {
+      exposureSlider.forceSetExposure(exposureSlider.getExposure());
+    });
```

Comment:
Thanks for remembering this. It seems like an incredibly rare occurrance to have the language direction changes while the exposure slider is displayed, but I guess it is the right thing to do to handle it.

The resizeHandler is added here in startEdit and removed in exitEdit(). With this change, you're calling ready() each time the user starts editing a new photo. I think if you add logging output at line 130, then edit photos a few times, and then switch the language, you'll find that the function is being executed once for each previous invocation of the editor.

Instead, you might need to add and remove event listeners for the 'localized' event